plenum/test/pool_transactions/helper.py

```diff
@@ -363,19 +367,26 @@ def sdk_add_new_node(looper,
     return nym_request, named_did
 
 
-async def prepare_node_request(new_node_name, steward_did, clientIp,
-                               clientPort, nodeIp, nodePort, bls_key, sigseed):
-    data = {
-        'alias': new_node_name,
-        'client_ip': clientIp,
-        'client_port': clientPort,
-        'node_ip': nodeIp,
-        'node_port': nodePort,
-        'services': ["VALIDATOR"],
-        'blskey': bls_key
-    }
-    nodeSigner = SimpleSigner(seed=sigseed)
-    destination = nodeSigner.identifier
+async def prepare_node_request(steward_did, new_node_name=None, clientIp=None,
+                               clientPort=None, nodeIp=None, nodePort=None, bls_key=None,
+                               sigseed=None, destination=None):
+    if (sigseed is None and destination is None):
+        raise AttributeError('You must specify destination directly or using sigseed')
+    if (sigseed is not None and destination is not None):
+        raise AttributeError('You must use only one of: sigseed or destination')
+    if (sigseed is not None):
+        nodeSigner = SimpleSigner(seed=sigseed)
+        destination = nodeSigner.identifier
+
+    data = {}
+    if new_node_name is not None: data['alias'] = new_node_name
+    if clientIp is not None: data['client_ip'] = clientIp
+    if clientPort is not None: data['client_port'] = clientPort
+    if nodeIp is not None: data['node_ip'] = nodeIp
+    if nodePort is not None: data['node_port'] = nodePort
+    if bls_key is not None: data['blskey'] = bls_key
+    data['services'] = ["VALIDATOR"]
```

Comment:
there are some tests require to demote node - i.e. set 'services' to empty array [ ], so you can extend this func for such a case

Reply:
Done